File: go/Calculator.go
package main

import (
	"fmt"
)

func add(a int, b int) int{
	return a + b
}

func sub(a int, b int) int{
	return a - b
}

func multiply(a int, b int) int{
	return a * b
}

func divide(a int, b int ) int{
	if b != 0 {
		return a / b
	} else {
		return 0
	}
}
func modulus(a int, b int ) int{
	if b != 0 {
		return a % b
	} else {
		return 0
	}
}

func main(){
	var a int
	var b int
	var opt string
	opts_map := map[string]int{"add": 1, "sub": 2, "mult": 3, "div": 4, "mod": 5}

	fmt.Print("Enter first integer: ")
	fmt.Scan(&a)
	fmt.Print("Enter Second integer: ")
	fmt.Scan(&b)
	fmt.Print("Enter option from the selections below:\nAdd: Enter add\nSubtract: Enter sub\nMultiply: Enter mult\nDivide: Enter div\nModulus/Remainder: Enter mod\nOption:")
	fmt.Scanln(&opt)
	switch opts_map[opt] {
	case 1:
		fmt.Printf("The addition of %d and %d is equal to %d",a,b,add(a,b))
	case 2:
		fmt.Printf("The subtraction of %d and %d is equal to %d", a,b, sub(a,b))
	case 3:
		fmt.Printf("The multiplication of %d and %d is equal to %d", a,b, multiply(a,b))
	case 4:
		fmt.Printf("The division of %d and %d is equal to %d", a,b, divide(a,b))
	case 5:
		fmt.Printf("The modulus of %d and %d is equal to %d", a,b, modulus(a,b))
	}
}

File: javascript/Calculator.js
const prompt = require("prompt-sync")();

class Calculator {
    constructor() { }

    add(a, b) {
        return a + b;
    }

    sub(a, b) {
        return a - b;
    }

    multiply(a, b) {
        return a * b;
    }

    divide(a, b) {
        if (b !== 0) {
            return a / b;
        } else {
            throw new Error("Not divisible by 0");
        }
    }

    modulus(a, b) {
        if (b !== 0) {
            return a % b;
        } else {
            throw new Error("Not divisible by 0");
        }
    }
}

const calc = new Calculator();
var opt = prompt("Enter an option from below:\nAdd: Enter add\nSubtract: Enter sub\nMultiply: Enter mult\nDivide: Enter div\nModulus/Remainder: Enter mod\nOption: ").trim().toLowerCase();
var a = parseInt(prompt("Enter first integer: "));
var b = parseInt(prompt("Enter second integer: "));

if (isNaN(a) || isNaN(b)) {
    console.log("Invalid input. Please enter valid integers.");
} else {
    try {
        let result;
        switch (opt) {
            case 'add':
                result = calc.add(a, b);
                break;
            case 'sub':
                result = calc.sub(a, b);
                break;
            case 'mult':
                result = calc.multiply(a, b);
                break;
            case 'div':
                result = calc.divide(a, b);
                break;
            case 'mod':
                result = calc.modulus(a, b);
                break;
            default:
                console.log("Invalid option. Please refresh and try again.");
                return;
        }

        if (result !== undefined) {
            console.log(`The result of ${opt} operation on ${a} and ${b} is: ${result}`);
        }
    } catch (error) {
        console.log(error.message);
    }
}
